### ISSUE TITLE
Does OSMOSE support detailed lanes tagging? 

### ISSUE THREAD

**Issue by matthewdarwin** [osm-fr/osmose-backend#573]
Re: https://wiki.openstreetmap.org/wiki/Lanes#Crossing_with_a_designated_lane_for_bicycles

Specifically does OSMOSE check:
- `turn:lanes`, `vehicle:lanes`, `bicycle:lanes` and `cycleway:lanes` have the same number of lanes (if the key is present)?
- `lanes` tag must have a count of less than or equal to the number of lanes in one of the previous tags?
- the values of all the lanes in the `:lanes` tags from a well-defined list (none,right,left,through,_empty_,etc..)
- check if `cycleway*` (`cycleway:left`, `cycleway:right`, `cycleway`=) and `turn:lanes` both exist then also `cycleway:lanes` must exist
- anything else relevant

Basically, before I start adding very complex tagging, I want to know if there are tools out there to validate what I am doing and find issues.   Editor support for this (eg JOSM) seems to be missing so I guess this is not commonly used?

https://taginfo.openstreetmap.org/keys/bicycle%3Alanes

**Comment by frodrigo** [comment]
I'm not sure, but a bicycle lane should not be counted.

Checking lanes is over complicated. Maybe can try to write some expected behaviours like in : https://github.com/osm-fr/osmose-backend/blob/master/plugins/Highway_Lanes.py#L239

**Comment by Thorwynn** [comment]
I have seen Osmose flag ways where the turn:lanes count does not match the number in the lanes=* key

**Comment by BalooUriza** [comment]
I disagree that bicycle lanes should not be counted.  It doesn't make much sense not to count them, especially since there's not a compelling reason not to map bicycle lanes as part of the rest of the lanes using access and turn lane tagging to integrate them like any other restricted lane.

**Comment by Famlam** [comment]
I think all of the initially requested things were implemented in the past few years as far as not debatable, so I'm closing this issue
